### DISCUSSION TITLE
Allow connections from non-loopback interfaces on macOS

### DISCUSSION THREAD

**Discussion by werediver** [lima-vm/lima#590] [Q&A] [answered]
After the issue #45 was resolved with the pull-request #283 (thanks to @AkihiroSuda, this is a great improvement for macOS!) Lima rejects connections from non-loopback interfaces (look for "pseudoloopback forwarder: rejecting non-loopback remoteAddr" cases in #283).

While this policy is a example of secure-by-default behavior, it would be beneficial to have control over it.

Would it be possible to introduce a VM configuration option for this? E.g. a top-level parameter `allowOnlyLoopbackConnections` (`true` by default) or a way to fine-tune `portForwards` entries.

I'm sure there is more than one use case for this, but currently I'm looking into connecting to a containerized service running on a macOS machine in Lima from a smartphone (mobile software development and debugging).

**Comment by jandubois** [comment, marked as answer]
You can do this with a simple `portForwards` rule:

```yaml
portForwards:
- guestPortRange: [1, 65535]
  hostIP: 0.0.0.0
```

Or create specific rules for the ports you want to expose externally.
